pkg/util/push/otel.go

```diff
@@ -150,10 +152,49 @@ func OTLPHandler(
 		)
 
 		req.Timeseries = metrics
+
+		if enableOtelMetadataStorage {
+			metadata := otelMetricsToMetadata(otlpReq.Metrics())
+			req.Metadata = metadata
+		}
+
 		return body, nil
 	})
 }
 
+func otelMetricsToMetadata(md pmetric.Metrics) []*mimirpb.MetricMetadata {
+	resourceMetricsSlice := md.ResourceMetrics()
+
+	metadataLength := 0
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		metadataLength += resourceMetricsSlice.At(i).ScopeMetrics().Len()
+	}
+	var metadata = make([]*mimirpb.MetricMetadata, metadataLength)
+	var metaDataPos = 0
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		resourceMetrics := resourceMetricsSlice.At(i)
+		scopeMetricsSlice := resourceMetrics.ScopeMetrics()
+
+		for j := 0; j < scopeMetricsSlice.Len(); j++ {
+			scopeMetrics := scopeMetricsSlice.At(j)
+			for k := 0; k < scopeMetrics.Metrics().Len(); k++ {
+				metric := scopeMetrics.Metrics().At(k)
+				entry := mimirpb.MetricMetadata{
+					Type:             mimirpb.MetricMetadata_MetricType(metric.Type()),
+					MetricFamilyName: prometheustranslator.BuildCompliantName(metric, "", true), // TODO expose addMetricSuffixes in configuration
```

Comment:
Is this TODO something you're planning to address in this PR?

Reply:
No, @gouthamve asked to leave the implementation of that for follow up PR here https://github.com/grafana/mimir/pull/5693#issuecomment-1669216300, should I remove the comment and create an issue for it?

Reply:
I'd create an issue yup.

Reply:
Created https://github.com/grafana/mimir/issues/5967. Should I remove the comment or leave it in?

Reply:
I'd put #5967 in the comment tbh. I'd leave it in.